utils/modular_model_converter.py

```diff
@@ -808,322 +1005,261 @@ def visit_ImportFrom(self, node: cst.ImportFrom) -> None:
+
+        # We need to replace the class node with the super class node
+        if len(bases) == 1:
+            super_class = bases[0]
+            super_file_name = self.model_specific_imported_objects[super_class]
+
+            # Get the mapper corresponding to the inherited class
+            mapper = self.visited_modules[super_file_name]
+            # Rename the super class according to the exact same rule we used when renaming the whole module
+            renamer = self.renamers[super_file_name]
+            renamed_super_class = preserve_case_replace(super_class, renamer.patterns, renamer.default_name)
+            renamed_super_class = convert_to_camelcase(renamed_super_class, renamer.old_name, renamer.default_old_name)
+
+            # Create the new class node
+            updated_node = replace_class_node(mapper, node, renamed_super_class)
+
+            # The node was modified -> look for all dependencies (recursively) of the new node
+            new_node_dependencies = ClassDependencyMapper.dependencies_for_new_node(updated_node, mapper)
+            all_dependencies_to_add = find_all_dependencies(
+                dependency_mapping=mapper.class_dependency_mapping,
+                initial_dependencies=new_node_dependencies,
+                initial_checked_dependencies=set(file_to_update.keys()),
             )
 
-        # Check if any new top-level function from the `modular_xxx.py` should be added to the different files
-        # (if it is called in a class in the file, then it will be copy pasted from `modular.py` to that file).
-        self._recursively_add_all_new_needed_functions_in_files()
+            relative_dependency_order = mapper.compute_relative_order(all_dependencies_to_add)
+            nodes_to_add = {
+                dep: (relative_dependency_order[dep], mapper.global_nodes[dep]) for dep in all_dependencies_to_add
+            }
 
-        for file, body in self.files.items():
+        # No super class, just check functions and assignments dependency in the imports from other model-specific files
+        else:
+            updated_node = node
+            # The node was NOT modified -> no need to look for dependencies recursively
+            all_dependencies_to_add = ClassDependencyMapper.dependencies_for_node(updated_node, self.global_nodes)
+
+            relative_dependency_order = self.compute_relative_order(all_dependencies_to_add)
+            nodes_to_add = {
+                dep: (relative_dependency_order[dep], self.global_nodes[dep])
+                for dep in all_dependencies_to_add
+                if dep not in file_to_update.keys()
+            }
+
+        # Add the class node itself to the nodes to add
+        class_idx = max(relative_dependency_order.values()) + 1 if len(relative_dependency_order) > 0 else 0
+        nodes_to_add[class_name] = (class_idx, updated_node)
+
+        return nodes_to_add, file_type
+
+    def create_modules(self) -> dict[str, cst.Module]:
```

Comment:
same here, if we create the modules post `leave_Module` I don't think this should be in this class no?

Reply:
Yes I was not sure about this choice, I kept it in the class because it was using a lot of internal variables. But as we move more and more stuff outside, let's keep a functional approach.